src/smc-webapp/project_new.cjsx

```diff
@@ -238,8 +239,10 @@ ProjectNewForm = rclass ({name}) ->
         e.preventDefault()
         if @state.filename[@state.filename.length - 1] == '/'
             @create_folder()
-        else
+        else if ReactDOM.findDOMNode(@refs.project_new_filename).value.includes('.')
```

Comment:
hmm, shouldn't this be just `@state.filename.includes('.')`? every change of the value in the box is automatically stored in the state, and it's also the state that's used for creating the filename. i.e. look into line 230:

```
create_file : (ext) ->
        @props.actions.create_file
            name         : @state.filename
```